tests/core/featurizers/test_tracker_featurizer.py

```diff
@@ -49,48 +55,1060 @@ def test_convert_labels_to_ids(domain: Domain):
+            },
+            {
+                'user': {'intent': 'mood_unhappy'}, 
+                'prev_action': {'action_name': 'utter_cheer_up'}
+            },
+            {
+                'user': {'intent': 'mood_unhappy'}, 
+                'prev_action': {'action_name': 'utter_did_that_help'}
+            },
+        ]
+    ]
+    if max_history is not None:
+        expected_states = [x[-max_history:] for x in expected_states]
+
+    assert actual_states is not None
+    assert len(actual_states) == len(expected_states)
+   
+    for actual, expected in zip(
+        actual_states, 
+        expected_states
+    ):
+        assert actual == expected
+
+
+@pytest.mark.parametrize("remove_duplicates", [True, False])
+@pytest.mark.parametrize("max_history", [None, 2])
+def test_multilabels_with_intent_max_history_tracker_featurizer(
+    moodbot_domain: Domain,
+    max_history: Optional[int],
+    remove_duplicates: bool
+):
+    state_featurizer = SingleStateFeaturizer()
+    tracker_featurizer = IntentMaxHistoryTrackerFeaturizer(
+        state_featurizer,
+        max_history=max_history,
+        remove_duplicates=remove_duplicates,
+    )
+
+    event_list1 = [
+        ActionExecuted("action_listen"),
+        user_uttered("greet"),
+        ActionExecuted("utter_greet"),
+        ActionExecuted("action_listen"),
+        user_uttered("mood_great"),
+    ]
+    tracker1 = DialogueStateTracker.from_events(
+        "default",
+        event_list1,
+        domain=moodbot_domain
+    )
+    event_list2 = [
+        ActionExecuted("action_listen"),
+        user_uttered("greet"),
+        ActionExecuted("utter_greet"),
+        ActionExecuted("action_listen"),
+        user_uttered("mood_unhappy"),
+    ]
+    tracker2 = DialogueStateTracker.from_events(
+        "default",
+        event_list2,
+        domain=moodbot_domain
+    )
+
+    _1, actual_labels, _2 = tracker_featurizer.featurize_trackers(
+        [tracker1, tracker2], moodbot_domain, RegexInterpreter()
+    )
+
+    greet_index = 5
+    mood_great_index = 6
+    mood_unhappy_index = 7
+
+    if remove_duplicates:
+        expected_labels = np.array([
+            [greet_index, -1],
+            [mood_great_index, mood_unhappy_index],
+            [mood_unhappy_index, mood_great_index],
+        ])
+    else:
+        expected_labels = np.array([
+            [greet_index, -1],
+            [mood_great_index, mood_unhappy_index],
+            [greet_index, -1],
+            [mood_unhappy_index, mood_great_index],
+        ])
+
+    assert actual_labels is not None
+    assert actual_labels.shape == expected_labels.shape
+    assert np.all(actual_labels == expected_labels)
+
+
+@pytest.mark.parametrize(
+    "tracker_featurizer_class", 
+    [
+        FullDialogueTrackerFeaturizer, 
+        MaxHistoryTrackerFeaturizer,
+        IntentMaxHistoryTrackerFeaturizer,
+    ]
+)
+def test_training_states_actions_and_entities_deprecated(
```

Comment:
Not sure we need tests for deprecation warnings... 🤔

Reply:
Yeah I can get rid of them, mostly just for myself to make sure that I had refactored everything correctly.